subprojects/core/src/integTest/groovy/org/gradle/api/SettingsDslIntegrationSpec.groovy

```diff
@@ -32,4 +32,67 @@ class SettingsDslIntegrationSpec extends AbstractIntegrationSpec {
         then:
         succeeds('help')
     }
+
+    def "Can access ExtensionAware in Groovy 'Settings'"() {
```

Comment:
:x: those added tests cover extensions in groovy, and, extra properties in various scenario in kotlin, they should be unified and uniform

Please make them two:
- `can type-safely use ExtensionAware with the Groovy DSL`
- `can type-safely use ExtensionAware with the Kotlin DSL`

Then, in each, assert on:
- extra properties
- extensions
- set from the script body
- set from the buildscript block
- set from an applied script

Reply:
So just add two tests that are exactly the same functionally for groovy and kotlin?

Reply:
Yep, and cover both DSLs